### PR TITLE
added Timer and HTML Parser

### PR DESCRIPTION
**Pull request by LStrouhal**
- added a Timer, which is displayed on the Trivia Game Page
- added a HTML parser to ensure special characters are loaded correctly in the app

### CODE REVIEW

File: frontend/src/components/Answers.js [hunk 1]
```diff
@@ -1,37 +1,53 @@
-import {useParams} from "react-router-dom";
-import {checkAnswer} from "../services/apiService";
-import {AnswerButtonStyle} from "./AnswersButtonStyle";
+import { useParams } from "react-router-dom";
+import { checkAnswer } from "../services/apiService";
+import { AnswerButtonStyle } from "./AnswersButtonStyle";
+import { useState } from "react";
+import parse from "html-react-parser";
```
Comment (dnnsjrng) [inline frontend/src/components/Answers.js:5]: 👍🏽

File: frontend/src/components/Questions.js [hunk 2]
```diff
@@ -5,20 +5,19 @@ import styled from "styled-components/macro";
...
     getSingleQuestion(questionID).then(setQuestionSet);
+    setSeconds(30);
   }, [questionID]);
 
-  if (!questionSet) {
-    return <p> loading </p>;
-  }
-
   const loadNextQuestion = () => {
     console.log({ numberOfQuestions, questionID });
```
Comment (dnnsjrng) [inline frontend/src/components/Questions.js:22]: _Could be removed_

File: frontend/src/components/Questions.js [hunk 3]
```diff
@@ -29,9 +28,25 @@ export default function Questions({ numberOfQuestions }) {
...
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (seconds >= 0) {
+        setSeconds(seconds - 1);
+        setVisibleSeconds(seconds);
+      } else {
+        loadNextQuestion();
+      }
+    }, 1000);
+    return () => clearTimeout(timer);
```
Comment (dnnsjrng) [inline frontend/src/components/Questions.js:40]: 👍🏽

File: frontend/src/components/Questions.js [hunk 4]
```diff
@@ -29,9 +28,25 @@ export default function Questions({ numberOfQuestions }) {
...
+  }, [seconds]);
+
+  if (!questionSet) {
+    return <p> loading </p>;
```
Comment (dnnsjrng) [inline frontend/src/components/Questions.js:44]: You could remove whitespace here.

File: frontend/src/components/Questions.js [hunk 5]
```diff
@@ -29,9 +28,25 @@ export default function Questions({ numberOfQuestions }) {
...
+  }
+
   return (
     <Wrapper>
-      <p>{questionSet.question}</p>
+      <p> {parse(questionSet.question)} </p>
```
Comment (dnnsjrng) [inline frontend/src/components/Questions.js:49]: You could remove whitespace here too.

File: frontend/src/pages/TriviaGame.js [hunk 6]
```diff
@@ -1,15 +1,28 @@
...
 
 export default function TriviaGame({ numberOfQuestions }) {
+  const [visibleSeconds, setVisibleSeconds] = useState(30);
+
   return (
     <>
-      <Header />
-      <Questions numberOfQuestions={numberOfQuestions} />
+      <Header>
+        <p id="seconds"> {visibleSeconds} seconds left</p>
```
Comment (dnnsjrng) [inline frontend/src/pages/TriviaGame.js:11]: You could remove the `id` tag here. This looks like a leftover from `getElemebtsById` era :)

**Review by dnnsjrng — APPROVED** [review]
👍🏽 Keep gambling! ▶️

File: frontend/src/pages/TriviaGame.js (hunk 6 above)
Comment (LStrouhal) [reply frontend/src/pages/TriviaGame.js:11]: Yep, correct, forgot to delete! Thanks!

**Pull request merged**